### PR TITLE
[DOCS] Improved Sorting syntax update ft.search.md and ft.aggregate.md

### PR DESCRIPTION
**Pull request by adrianoamaral**
adding argument `WITHCOUNT` 
explain the behaviour of SORT in `DIALECT 4`

**Describe the changes in the pull request**

A clear and concise description of what the PR is solving, including:
1. The current state briefly
2. What is the change
3. Adding the outcome (changed state)

**Which issues this PR fixes**
1. #...
2. MOD...

**Main objects this PR modified**
1. ...
2. ...

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: docs/commands/ft.aggregate.md [hunk 1]
```diff
@@ -74,7 +74,15 @@ sorts the pipeline up until the point of `SORTBY`, using a list of properties.
...
+
+**Sorting Optimisations**: performance is optimised for sorting operations on `DIALECT 4` in different scenarios:
+   - Skip Sorter - applied when there is no sort of any kind. The query can return once it reaches the `LIMIT` requested results.
+   - Partial Range - applied when there is a `SORTBY` a numeric field, with no filter or filter by the same numeric field, the query iterate on a range large enough to satisfy the `LIMIT` requested results.
+   - Hybrid - applied when there is a `SORTBY` a numeric field in addition to another non-numeric filter. Some results will get filtered, and the initial range may not be large enough. The iterator then is rewinded with the following ranges, and an additional iteration takes place to collect `LIMIT` requested results.
+   - No optimization - If there is a sort by score or by non-numeric field, there is no other option but to retrieve all results and compare their values.
+
+**Counts behaviour**: optional`WITHCOUNT`argument return accurate counts for the query results with sorting. This operation processes all results in order to get accurate count, being less performant than the optimised option (default behaviour on `DIALECT 4`)
```
Comment (DvirDukhan) [inline docs/commands/ft.aggregate.md:85]:
```suggestion
Attributes needed for `SORTBY` should be stored as `SORTABLE` to be available with very low latency.

**Sorting Optimizations**: performance is optimized for sorting operations on `DIALECT 4` in different scenarios:
   - Skip Sorter - applied when there is no sort of any kind. The query can return once it reaches the `LIMIT` requested results.
   - Partial Range - applied when there is a `SORTBY` clause over a numeric field, with no filter or filter by the same numeric field, the query iterate on a range large enough to satisfy the `LIMIT` requested results.
   - Hybrid - applied when there is a `SORTBY` clause over a numeric field and another non-numeric filter. Some results will get filtered, and the initial range may not be large enough. The iterator is then rewinding with the following ranges, and an additional iteration takes place to collect the `LIMIT` requested results.
   - No optimization - If there is a sort by score or by non-numeric field, there is no other option but to retrieve all results and compare their values.

**Counts behavior**: optional `WITHCOUNT` argument returns accurate counts for the query results with sorting. This operation processes all results in order to get an accurate count, being less performant than the optimized option (default behavior on `DIALECT 4`)

```

File: docs/commands/ft.search.md [hunk 2]
```diff
@@ -171,9 +171,18 @@ adds an arbitrary, binary safe payload that is exposed to custom scoring functio
...
+
+**Sorting Optimisations**: performance is optimised for sorting operations on `DIALECT 4` in different scenarios:
+  - Skip Sorter - applied when there is no sort of any kind. The query can return once it reaches the `LIMIT` requested results.
+  - Partial Range - applied when there is a `SORTBY` a numeric field, with no filter or filter by the same numeric field, the query iterate on a range large enough to satisfy the `LIMIT` requested results.
+  - Hybrid - applied when there is a `SORTBY` a numeric field in addition to another non-numeric filter. Some results will get filtered, and the initial range may not be large enough. The iterator then is rewinded with the following ranges, and an additional iteration takes place to collect `LIMIT` requested results.
+  - No optimization - If there is a sort by score or by non-numeric field, there is no other option but to retrieve all results and compare their values.
+
+**Counts behaviour**: optional`WITHCOUNT`argument return accurate counts for the query results with sorting. This operation processes all results in order to get accurate count, being less performant than the optimised option (default behaviour on `DIALECT 4`)
+
```
Comment (DvirDukhan) [inline docs/commands/ft.search.md:185]:
```suggestion

**Sorting Optimizations**: performance is optimized for sorting operations on `DIALECT 4` in different scenarios:
  - Skip Sorter - applied when there is no sort of any kind. The query can return once it reaches the `LIMIT` requested results.
  - Partial Range - applied when there is a `SORTBY` clause over a numeric field, with no filter or filter by the same numeric field, the query iterate on a range large enough to satisfy the `LIMIT` requested results.
  - Hybrid - applied when there is a `SORTBY` clause over a numeric field and another non-numeric filter. Some results will get filtered, and the initial range may not be large enough. The iterator is then rewinding with the following ranges, and an additional iteration takes place to collect the `LIMIT` requested results.
  - No optimization - If there is a sort by score or by non-numeric field, there is no other option but to retrieve all results and compare their values.

**Counts behavior**: optional`WITHCOUNT`argument returns accurate counts for the query results with sorting. This operation processes all results in order to get an accurate count, being less performant than the optimized option (default behavior on `DIALECT 4`)


```

**Review by DvirDukhan — CHANGES_REQUESTED** [review]

**Review by DvirDukhan — APPROVED** [review]

File: docs/commands/ft.aggregate.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ syntax: |
     [LOAD count field [field ...]] 
     [TIMEOUT timeout] 
     [ GROUPBY nargs property [property ...] [ REDUCE function nargs arg [arg ...] [AS name] [ REDUCE function nargs arg [arg ...] [AS name] ...]] ...]] 
-    [ SORTBY nargs [ property ASC | DESC [ property ASC | DESC ...]] [MAX num]] 
+    [ SORTBY nargs [ property ASC | DESC [ property ASC | DESC ...]] [MAX num] [WITHCOUNT]
```
Comment (oshadmi) [inline docs/commands/ft.aggregate.md:9]: @adrianoamaral `WITHCOUNT` and `WITHOUTCOUNT` can also be used without sorting
The location of these arguments should be related to the command itself.
Related #3651

File: docs/commands/ft.aggregate.md [hunk 4]
```diff
@@ -74,7 +74,16 @@ sorts the pipeline up until the point of `SORTBY`, using a list of properties.
...
+   - Skip Sorter - applied when there is no sort of any kind. The query can return once it reaches the `LIMIT` requested results.
+   - Partial Range - applied when there is a `SORTBY` clause over a numeric field, with no filter or filter by the same numeric field, the query iterate on a range large enough to satisfy the `LIMIT` requested results.
+   - Hybrid - applied when there is a `SORTBY` clause over a numeric field and another non-numeric filter. Some results will get filtered, and the initial range may not be large enough. The iterator is then rewinding with the following ranges, and an additional iteration takes place to collect the `LIMIT` requested results.
+   - No optimization - If there is a sort by score or by non-numeric field, there is no other option but to retrieve all results and compare their values.
+
+**Counts behavior**: optional `WITHCOUNT` argument returns accurate counts for the query results with sorting. This operation processes all results in order to get an accurate count, being less performant than the optimized option (default behavior on `DIALECT 4`)
```
Comment (oshadmi) [inline docs/commands/ft.aggregate.md:85]:
```suggestion
**Counts behavior**: optional `WITHCOUNT` argument returns accurate counts for the query results with sorting. This operation processes all results in order to get an accurate count, being less performant than the `WITHOUTCOUNT` option (default behavior on `DIALECT 4`)

```

**Review by oshadmi — COMMENTED** [review]
Few comments

File: docs/commands/ft.aggregate.md (hunk 3 above)
Comment (adrianoamaral) [reply docs/commands/ft.aggregate.md:9]: In this case, we won't mentioning WITHTOUTCOUNT

**Review by adrianoamaral — COMMENTED** [review]
for now, we won't have mention to WIHTOUTCOUNT behaviour beside being default on DIALECT 4

**Pull request merged**